File: RR.AI-Chat/RR.AI-Chat.Dto/SessionDto.cs
namespace RR.AI_Chat.Dto
{
    public class SessionDto
    {
        public Guid Id { get; set; }

        public Guid? ProjectId { get; set; }

        public string Name { get; set; } = null!;

        public DateTimeOffset DateCreated { get; set; }

        public DateTimeOffset DateModified { get; set; }
    }

    public class SessionConversationDto : SessionDto
    {
        public List<SessionMessageDto> Messages { get; set; } = [];
    }
}

File: ai-chat-ui/src/app/dtos/SessionDto.ts
export interface SessionDto {
  id: string;
  name: string;
  projectId?: string;
  dateCreated: string; // ISO date string from API
  dateModified: string; // ISO date string from API
}

export interface SessionMessageDto {
  id: string;
  name: string;
  dateCreated: string;
  dateModified: string;
  text: string;
  role: number;
}

export interface SessionConversationDto {
  id: string;
  name: string;
  dateCreated: string;
  dateModified: string;
  messages: SessionMessageDto[];
}
